File: examples/php/advanced/02_oop.php
<?php
// PHP Advanced: OOP

class Animal
{
    public $name;

    public function __construct($name)
    {
        $this->name = $name;
    }

    public function speak()
    {
        return "...";
    }
}

class Dog extends Animal
{
    public function speak()
    {
        return "Woof!";
    }
}

$dog = new Dog("Buddy");
echo $dog->name . " says " . $dog->speak() . "\n";
?>

File: public/examples/ruby/advanced/02_oop.rb
# Ruby Advanced: OOP

class Animal
  attr_accessor :name
  
  def initialize(name)
    @name = name
  end
  
  def speak
    "..."
  end
end

class Dog < Animal
  def speak
    "Woof!"
  end
end

dog = Dog.new("Buddy")
puts "#{dog.name} says #{dog.speak}"
